curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/65672136535209761a5cf02b.md

```diff
@@ -13,10 +13,22 @@ Don't forget you need to interpolate with the dollar sign here.
 
 # --hints--
 
-You should add an `onclick` with the value `playSong(${song.id})` as the second attribute and value of the `button` element.
+You do not need to change the other parts of the code. All you need to do is add the `onclick` attribute to the button element.
 
 ```js
-assert.match(code, /<button\s+class=('|")playlist-song-info\1\s*onclick=\1playSong\(\$\{song\.id\}\)\1>\s*<span\s+class=\1playlist-song-title\1>\$\{song\.title\}<\/span>\s*<span\s+class=\1playlist-song-artist\1>\$\{song\.artist\}<\/span>\s*<span\s+class=\1playlist-song-duration\1>\$\{song\.duration\}<\/span>\s*<\/button>/)
+assert.match(code, /<button\s.*>\s*<span\s+class=('|")playlist-song-title\1>\$\{song\.title\}<\/span>\s*<span\s+class=\1playlist-song-artist\1>\$\{song\.artist\}<\/span>\s*<span\s+class=\1playlist-song-duration\1>\$\{song\.duration\}<\/span>\s*<\/button>/)
```

Comment:
@jdwilkin4 Are we not able to test the document object? E.g. `assert.includes(document.querySelector("button").classList, "playlist-song-title")`.

Reply:
> Are we not able to test the document object?

I wasn't aware there was an issue with that. That should work

Reply:
Testing that everything remains the same seems out of scope for this step. If we had to do that every time our tests would become bloated.

The test message also becomes unclear, "something that shouldn't change, changed" isn't a very helpful message. The regex is just too large for a single assert message and it won't provide useful feedback.

I think all we do here is check that the class remains on the element after the handler is added.